### ISSUE TITLE
Product price inclusive of tax is adding tax on the checkout page

### ISSUE THREAD

**Issue by tsetend** [woothemes/woocommerce#36221]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The product price is inclusive of tax. The price is displayed correctly on the product page however on the cart and checkout page additional tax is added on the product price.

Example: We have a product A with a total price of £375. On the cart and checkout page the total price becomes £450.00

Tax is applied to UK for 20% only. Tax is not applied on any other countries

Issue originally reported for a user in 5820904-zen

### Expected behavior

The price on the cart as well as the checkout page should remain the same i.e £375

### Actual behavior

The price of the product changes to £450.00 instead

### Steps to reproduce

1. Setup tax **Prices entered with tax** | Yes, I will enter prices inclusive of tax.
![screenshot](https://d.pr/i/AdeErL+) **Image Link:** https://d.pr/i/AdeErL
2. Use the standard tax class with 20% VAT for UK customers.
![screenshot](https://d.pr/i/LROIfG+) **Image Link:** https://d.pr/i/LROIfG
3. Add a product with £375 price
![screenshot](https://d.pr/i/uMnTXm+) **Image Link:** https://d.pr/i/uMnTXm
5.  Product price shows fine here:
![screenshot](https://d.pr/i/YHezlc+) **Image Link:** https://d.pr/i/YHezlc
6. Cart and checkout page shows price exclusive of tax:
![screenshot](https://d.pr/i/nhlT24+) **Image Link:** https://d.pr/i/nhlT24

 

### WordPress Environment

Tested on my local test environment with the latest version of WP and Woo. Using Storefront theme.

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by kevinruscoe** [comment]
I think the issue is with step 3 n your screengrabs. You're entering your prices inclusive of VAT, then marking the item as "taxable", when it should really be "None" as there's no tax to be applied.

I can certainly see why this looks misleading though.  There's 2 solutions to this

1 - We remove the "Taxable" option from the tax status option on the products, somehow letting admins know WHY you can't select the option.
2 - Ignore the tax status when calculating the prices

**Comment by kevinruscoe** [comment]
Adding 

```
if ('yes' === get_option( 'woocommerce_prices_include_tax' )) {
    return 'none';
}
```

To WC_Product->is_taxable() (abstract-wc-product.php:L1680) "fixes" this. This line just ignores the prop if the global setting is defined.

There's a filter here as well, so you could add this "fix" in your theme.

```
add_filter( 'woocommerce_product_is_taxable', function( $is_taxable, $product ){
	if ('yes' === get_option( 'woocommerce_prices_include_tax' )) {
		$is_taxable = false;
	}

	return $is_taxable;
}, 20, 2);
```

The filter also passes the $product, so you could do more checks here to triple check if you want to remove tax.

**Comment by tsetend** [comment]
@kevinruscoe Thank you for the update. 

I don't know what changed in my local site as I don't remember changing anything. It is the same site where I first was able to replicate the behavior but now I cannot see this problem:

![screenshot](https://d.pr/i/Nsvj08+) **Image Link:** https://d.pr/i/Nsvj08

I looked at the customer's site but I see the problem and will recommend to change the tax status to none for now. But ideally it should show the product price + VAT charge like the screenshot above. If we change the tax to none, it will only show the product price of £375.

**Comment by johndcoy** [comment]
@kevinruscoe What I believe is happening is some users expect to see a tax calculation on the checkout page when "Prices entered with tax" is inclusive of tax, **without that tax line item adding to the total cost**. Since users would like to use the reports feature when reviewing taxes setting "none" is not an option.

How can users setting prices inclusive of tax see the calculation of taxes on the checkout page or reports?

Another instance 5825512-zen
The user sets prices inclusive of tax and expects to see the price remain 140 regardless of location, while still showing the calculation.

**Comment by shevene** [comment]
I believe what we're seeing here is the [default behavior for WooCommerce](https://github.com/woocommerce/woocommerce/wiki/How-Taxes-Work-in-WooCommerce#prices-including-tax---experimental-behavior) when prices are set to be inclusive of tax. For example, if a store is based in the United States, then you wouldn't be expected to collect taxes from any other country. Thus, someone from another country would have taxes for the US subtracted from the original price, and then whatever custom tax rate that has been set for their country would then be added to that new price.

By using this snippet, you can override that behavior:

`add_filter( 'woocommerce_adjust_non_base_location_prices', '__return_false' );`

**Comment by peterfabian** [comment]
I'm sorry, this issue seems a bit confusing. Looking at the support request, perhaps the customer wanted to display the price breakdown between product price and VAT on the 
- product page, 
- cart, and 
- in the checkout 

While using the following settings:
- Yes, I will enter prices inclusive of tax
- Display prices in the shop: Including Tax
- Display prices during cart and checkout: Including Tax

I think there are some partial solutions like setting the price display suffix to `({price_excluding_tax} + tax)`, but I believe this won't help with Cart or Checkout where one might need to use custom templates atm. So this feels more like a feature request than a bug. Would you agree?

**Comment by jessepearson** [comment]
Just noting that I came across this after I encountered similar behavior. My issue was that my store address was not set to the proper base country. In my instance the base tax was based off of the country code CH, but my store country was US. Once I updated the store country to CH, it displayed as expected.

Basically, saying that if you're expecting certain tax rates to be applied, your store address needs to match the tax rate settings.

**Comment by JohnyBravo456** [comment]
@meetshevene That code doesnt seem to work. Can you explain how you added it in.

**Comment by jessepearson** [comment]
> @meetshevene That code doesnt seem to work. Can you explain how you added it in.

You would add that either in the functions.php file of the theme, or through a plugin like [Code Snippets.](https://wordpress.org/plugins/code-snippets/)

**Comment by JohnyBravo456** [comment]
@jessepearson yup Did that and nothing happens. Im baffled as to why. We use variable products, which someone else said doesnt work with this code, so we tested it with a simple product and it still does not apply. We tried smacking it into the functions.php - nothing, tried snippets - nothing, disabled caching temporarily - nothing. 

From a similar post:

Are there any tax related settings that need to be setup to impact this code? Other snippets seem to work fine, just this one does nothing

To clarify: If I set taxes to inclusive and shop location to germany(which is necessary for this code):
German customers get the Base price (inc 19%tax) but total cost is the base value
Italian customers get the base price - german taxes + italian taxes giving them a total cost of 112 instead of 100 for example
Non EU customers get the base price -19% +0% and have a total cost of 80.

It does not matter if I add the code in code snippets or if I put it directly into functions.php. There is no change.

Im assuming this is the exact issue talked about here. The base price should stay the same and show in the checkout
65$ (inc 19% Vat) for german customers
65$ inc. 24% Vat) for greece customers
65$ inc 0% Vat) for US customers

**Comment by JohnyBravo456** [comment]
OK UPDATE: because I hate it when people say they fixed it but never say how: I figured It must be some conflict with our plugins since any and all ways of modifing tax data had no effect on our site. I calculated what prices should bee if the customer is in germany since our shop location is in germany and that was the baseline (also I tinkered with the idea of overcoming the issue by creating price adjusted zones and that calc. was the first of the zones) and figured that our home tax calc was running not on 19% as it should but on 16%, a rate that is both niche and also in none of all these tax plugins / tables or data discussed here. Pinpointed it to an old plugin (One stop shop for woocommerce) and turned it off. Fixed everything.

TLDR: Plugin caused a tax calc haywire. Fixed it by disabling, One stop shop plugin, did not use code above in functions php or snippet, instead downloaded EU/UK VAT Compliance for WooCommerce (Free) and set -> settings -> WooCommerce VAT settings (new settings from the VAT compliance plugin) -> Same net prices everywhere CHECKMARK 

10 hours and 76 tabs later site is back up and running.

**Comment by kloon** [comment]
From the looks of things, this sounds more like a configuration issue or a conflict as has been indicated by others. I am going to close this issue, but happy to re-open should a valid test case be presented to help us recreate this as described.
